src/Service/ServiceManagerConfig.php

```diff
@@ -150,6 +150,7 @@ public function __construct(array $config = [])
                 // implementations; they're dealt with later.
                 if ($instance instanceof ServiceLocatorAwareInterface
                     && ! $instance instanceof AbstractPluginManager
+                    && ! $instance->getServiceLocator()
```

Comment:
The changes in the 2.7 branch are for the following reasons:

- We test for plugin managers, because we want to vary the deprecation message: plugin managers need the parent locator passed during instantiation in order to be forwards compatible, while non-plugin manager, ServiceLocatorAware instances should be updated to have their dependencies injected discretely.
- The original behavior was, as noted, to always inject, without regard for whether or not an instance was present. This is *still* true, except in the case of abstract plugin managers. There is no BC break.

The reason for the change with regard to abstract plugin managers is that all of those managed by ZF itself have been updated at this point to inject the parent locator during instantiation, and we do not want to raise the deprecation notice for them, as they're already updated. The new code block is literally for *custom* plugin managers that are registered by end-users, to notify them they need to update their factories. In fact, it was introduced because they were *not* getting injected by previous iterations in the 2.7 series, which was causing errors.

@svycka I guess what I'm wondering is: what exact messages are you getting from ZfcUser? Can you share the message and stack trace? Are there any custom plugin managers in ZfcUser? If so, where are the factories for them?